src/site/topics/objects-review/objects-review.rst

```diff
@@ -494,412 +502,320 @@ References
-* Either way, when we add the ``newFriend`` to our array, we are now guaranteed to have room
+* Either way, the ``Friend`` being added via the ``add`` method will always go to the next available spot
+* When done, this method returns a ``boolean`` indicating if the ``add`` worked correctly
 
-* You will also see that the ``expandCapacity`` method is ``private``
-    * This method is important for the inner workings of the ``ContactList`` class
-    * This method is not something I want the user of my class to care about
-        * Abstraction
+* Also notice that the ``expandCapacity`` method is ``private``
 
+    * This method is important for the inner workings of the ``ContactList`` class
+    * This method is not something one wants a user of this ``ContactList`` class to care about
 
-Remove Friends
-^^^^^^^^^^^^^^
 
-**Java**
+Contains and Find
+^^^^^^^^^^^^^^^^^
 
-.. code-block:: java
+.. literalinclude:: /../main/java/ContactList.java
+    :language: java
     :linenos:
-    :emphasize-lines: 8, 11
-
-        public void remove(String firstName, String lastName) {
-            // Create a temp friend object for easy
-            // use of the Friend class' equals()
-            Friend toDelete = new Friend(firstName, lastName, "");
-
-            // Linear search for the friend we are trying to delete
-            for (int i = 0; i < friendCount; ++i) {
-                if (toDelete.equals(friends[i])) {
-                    // Have friend at the end of the array be referenced
-                    // by the array index we removed from
-                    friends[i] = friends[friendCount - 1];
-                    friends[friendCount - 1] = null;
-                    friendCount--;
-                }
-            }
-        }
-
-* We'll create a temporary ``Friend`` based on the parameters so we can make use of our ``Friend`` class' ``.equals``
-* To remove the ``Friend`` all we need to do it lose reference to it
-    * Garbage
-* In the above example, we made the array at the index of the removed ``Friend`` to reference the ``Friend`` at the end of the array
-    * ``friendCount - 1``
-* After this, the array has no reference to the ``Friend`` that was removed
+    :lineno-start: 69
+    :lines: 69-95
 
-* You may notice that the array at index ``friendCount - 1`` is set to ``null``
-    * This is done to deal with the edge case of removing the last friend
 
-* You may also notice that this method, as it is written, will not do anything special if the ``Friend`` we try to remove does not exist
-* What should we do in this scenario?
-    * Ignore it?
-    * Return a boolean?
-    * Crash the program?
-    * Explode?
+* The ``find`` method, which is ``private``, is an internal helper method for finding the index of a given ``Friend``
+* This method is just a linear search
 
-* This will be discussed further a little later in the course
+    * Mind the use of ``Objects.equals`` which is a null safe way to check if two objects are equal
```

Comment:
Maybe worth reinforcing, _...if two objects are equal via their `equals` method_.

Reply:
clarity is clarity